### DISCUSSION TITLE
Angular + @tailwind recompiles scss on every change

### DISCUSSION THREAD

**Discussion by urugator** [tailwindlabs/tailwindcss#11308] [Help]
Hi, I am using Tailwind 3 with Angular 16. 
Normally, when I don't touch any `scss` file during development, angular does NOT recompile these. 
However when I include any `@tailwind` directive into my `scss` file, the styles are recompiled on every file change, even when I modify only `.ts` sources. This is unfortunate, because compiling styles takes significantly longer than incremental build of `ts` code (~10s VS ~1s).

Is this expected behavior?
Is there anything I can do about it?

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: {
    relative: true,
    files: [`./src/**/*.{html,ts}`],
  },
  theme: {
    // Material often sets only "color" assuming "border-color" will use it as well (as by spec).
    // However preflight resets border-color to theme('borderColor.DEFAULT', currentColor)
    // and default theme sets 'borderColor.DEFAULT' to theme('colors.gray.200', 'currentColor').
    // So we just want to prevent borderColor.DEFAULT being set by default theme.
    borderColor: ({ theme }) => theme('colors'),
    extend: {
      colors: {
        text: { DEFAULT: 'redacted' },
        primary: { DEFAULT: 'redacted' },
        secondary: { DEFAULT: 'redacted' },
        accent: { DEFAULT: 'redacted' },
        warn: { DEFAULT: 'redacted' },
        success: { DEFAULT: 'redacted' },
      },
    },
  },
  plugins: [],
};
```

**Comment by wongjn** [comment]
> Is this expected behavior?

I would say so, since Tailwind works by reading the source code configured by `content` and producing CSS rules from this. So if you change said source code, Tailwind would need to rescan the code and (re)generate the CSS rules.

> Is there anything I can do about it?

You could consider:
- Being less broad with your `content` file globs.
- Separate out Tailwind from your Sass build pipeline.

**Reply by urugator** [reply]
Thank you for the answer. The biggest slowdown are `@angular/material` styles, so I will take a look if I can isolate it's compilation from tailwind. The generation of tailwind classes isn't that slow I think - however I wonder if it wouldn't be faster to simply emit all classes for devel. I guess I will end up with multiple scss entry points compiled to different `css` files.
